src/content/docs/en/guides/troubleshooting.mdx

```diff
@@ -132,6 +132,26 @@ Yarn 2+, a.k.a. Berry, uses a technique called [Plug'n'Play (PnP)](https://yarnp
 nodeLinker: "node-modules"
 ```
 
+### Using Astro in a monorepo
+
+When working with Astro in a monorepo setup, it is recommended to keep all the dependencies in the project `package.json` file.
+However, you may also want to use Astro in the root of the monorepo (for example, to generate static routes properly). In this case, you may encounter some issues with dependencies not being installed and bundled properly.
+To avoid this, you need to manually add these dependencies to the `vite.ssr.noExternal` part of Astro's config:
```

Comment:
```suggestion
When working with Astro in a monorepo setup, project dependencies should be added in each project's own `package.json` file.

However, you may also want to use Astro in the root of the monorepo (e.g. to generate static routes properly). In this case, manually add these dependencies to the `vite.ssr.noExternal` part of Astro's config to ensure that these dependencies are properly installed and bundled:
```
Just my take at tightening up the wording.

Reply:
> (for example, to generate static routes properly)

I'm not sure if this is the reason to install Astro at the root from the linked issue. From https://github.com/withastro/astro/issues/6790#issuecomment-1503518727, it's a convention by Nx? So we could say, 

> e.g. [Nx](https://nx.dev) projects recommend installing dependencies at the root

Also, is there a link that shows Nx suggesting it?

---


> manually add these dependencies to the `vite.ssr.noExternal` part of Astro's config

Maybe we can explain the kind of dependencies required to be added, which are packages that rely on Astro, e.g. `@astrojs/cloudflare`, `@astrojs/vue`, `astro-component-lib`. So on top of Sarah's suggestion, it could be

> In this case, manually add Astro-related dependencies to the ...